src/lib.rs

```diff
@@ -362,6 +372,37 @@ where
                 RawColors::<RawU32>::new(&self.raw_bmp)
                     .map(|raw| Rgb888::from(RawU24::new(raw.into_inner())).into()),
             ),
+            ColorType::Argb8888 => {
+                target.fill_contiguous(
+                    &area,
+                    RawColors::<RawU32>::new(&self.raw_bmp).map(|raw| {
+                        // integer blending approach from https://stackoverflow.com/a/12016968
+                        let v = raw.into_inner();
+                        let mut alpha = v & 0xFF;
+                        let inv_alpha = 256 - alpha;
+                        alpha += 1;
+                        if alpha == 0 {
+                            // pixel is completely transparent, use bg color
+                            self.alpha_bg
+                        } else if alpha == 255 {
+                            // pixel is completely opaque, just use its color
+                            Rgb888::from(RawU24::new(v >> 8))
```

Comment:
I didn't have time to thoroughly check the blending code, but the checking for `alpha == 0` and `alpha == 255` must happen before `1` is added.